st3/sublime_lib/pure_resource_path.py

```diff
@@ -0,0 +1,103 @@
+import posixpath
+from functools import total_ordering
+
+
+__all__ = ['PureResourcePath']
+
+
+@total_ordering
+class PureResourcePath():
+    """A representation of a Sublime Text resource path. Inspired by pathlib."""
+
+    def __init__(self, *pathsegments):
+        self._parts = tuple(
+            part
+            for segment in pathsegments if segment
+            for part in posixpath.normpath(str(segment)).split('/')
+        )
+
+    def __hash__(self):
+        return hash(self.parts)
+
+    def __repr__(self):
+        return "%s(%s)" % (type(self).__name__, repr(str(self)))
+
+    def __str__(self):
+        return '/'.join(self.parts)
+
+    def __eq__(self, other):
+        return self._parts == other.parts
+
+    def __lt__(self, other):
+        return self._parts < other.parts
+
+    def __truediv__(self, other):
+        return self.joinpath(other)
+
+    def _new(self, *args):
```

Comment:
Remove this and replace usages with `self.__class__(*args)` imo.

Reply:
Done.